R/execution.r

```diff
@@ -83,12 +62,13 @@ execute = function(request) {
     payload <<- lappend(payload, list(source='page', text=paste(text, collapse="\n")))
   })
 
-  send_plot = function(plotobj) {
-      tf = tempfile(fileext='.png')
-      do.call(png, c(list(filename=tf), get_plot_options()))
-      replayPlot(plotobj)
-      dev.off()
-      display_png(file=tf)
+  send_plot <- function(plotobj) {
+      params <- list()
+      for (mime in getOption('jupyter.plot_mimetypes')) {
+          r <- mime2repr[[mime]](plotobj)
+          params[[mime]] <- if (is.raw(r)) base64encode(r) else r
+      }
+      do.call(display_alternatives, params)
```

Comment:
This is already the case - IRdisplay provides the display API, that I also want to make usable from the R magic in a Python notebook, so the same code works in both. However, it doesn't know how to actually display the data - it needs to be run by something that gives it a base display function to send the data. IRkernel does this. It's not very nice, but it seems to work.

I'm still not convinced that display_alternatives is necessary, though.